test/integration/test_utils.mjs

```diff
@@ -291,35 +306,13 @@ async function pasteFromClipboard(page, data, selector, timeout = 100) {
     await navigator.clipboard.write([new ClipboardItem(items)]);
   }, data);
 
+  const validator = e => e.clipboardData.items.length !== 0;
   let hasPasteEvent = false;
   while (!hasPasteEvent) {
     // We retry to paste if nothing has been pasted before the timeout.
```

Comment:
Won't this loop now continue indefinitely if the timeout is reached, since `waitForEvent` now returns `null` in that case?